### PR TITLE
Annotation quick fixes for attribute inspections

### LINKED ISSUE #4672: The attribute related inspections could use annotation sided quickfixes
Currently, all quickfixes for the `MissingAttributeInspection`, `AttributeValueOutOfSyncInspection`, `MissingModuleAnnotationInspection` and `MissingMemberAnnotationInspection` manipulate the attributes, but none the annotations. In particular for the last two inspections this is not very satisfying.

I propose to add a `RemoveAnnotationQuickFix`, an `AdjustAnnotationQuickFix` and an `AddAttributeAnnotationQuickFix`. Here, the latter two should take into consideration that there are specific annotations for special attribute/value combinations.

### PR DESCRIPTION
**Pull request by MDoerner**
This PR closes #4672

This PR introduces the three quickfixes mentioned in the PR above. This completes the set necessary to handle attributes via the a annotation/inspection/quickfix system.

In addition, this PR centralizes the functionality to add, remove and update/replace annotations in one place.

### CODE REVIEW

File: RubberduckTests/QuickFixes/AdjustAttributeAnnotationQuickFixTests.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Attribute Foo.VB_UserMemId = -40
+    Const const1 As Integer = 9
+End Sub";
+
+            const string expectedCode =
+                @"'@MemberAttribute VB_UserMemId, -40
+Public Sub Foo()
+Attribute Foo.VB_UserMemId = -40
+    Const const1 As Integer = 9
+End Sub";
```
[the remark this answers is not in the document]
Comment (MDoerner) [reply RubberduckTests/QuickFixes/AdjustAttributeAnnotationQuickFixTests.cs]: I do not see why this is confusing. These values mean totally different things. It is just an implementation detail that they use the same attribute.

I think for a programmer without any knowledge of COM, it is more intuitive to simply have `DefaultMember` and `Enumerator` annoatations.

File: RubberduckTests/PostProcessing/AnnotationUpdaterTests.cs [hunk 2]
```diff
@@ -0,0 +1,749 @@
...
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor.SafeComWrappers;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+using RubberduckTests.Mocks;
+
+namespace RubberduckTests.PostProcessing
+{
+    [TestFixture]
+    public class AnnotationUpdaterTests
```
[the remark this answers is not in the document]
Comment (MDoerner) [reply RubberduckTests/PostProcessing/AnnotationUpdaterTests.cs]: The `AnnotationUpdater` only deals with annotations. The `AttributesUpdater`, which has been added a few PRs back, deals with attributes.

File: RubberduckTests/QuickFixes/AdjustAttributeAnnotationQuickFixTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bclothier) [reply RubberduckTests/QuickFixes/AdjustAttributeAnnotationQuickFixTests.cs]: But wouldn't the same person be wondering why the annotation changed completely from `@DefaultMember` to `@MemberAttribute VB_UserMemId, -40`? Furthermore, what about the `@MemberAttribute VB_UserMemId, 0` ? That'd be a duplicate (and a poor one at that). I agree that `@DefaultMember` is more intuitive and succinct, but considering that it's just a one special value of the `VB_UserMemId`, a `@MemberId` annotation with constant names seems to me a better general solution to encourage users to treat it in a consistent. 

An added benefit is that it also hints that you cannot do `@DefaultMember, @Enumerator` at the same time, eliminating another potential misuse of the annotations. However, in the end, it's going to be out of the scope for this PR. I just wanted to point this behavior out.

File: RubberduckTests/QuickFixes/AddAttributeAnnotationQuickFixTests.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+using Rubberduck.Inspections.Concrete;
+using Rubberduck.Inspections.QuickFixes;
+using Rubberduck.Parsing.Annotations;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.VBA;
+
+namespace RubberduckTests.QuickFixes
+{
+    [TestFixture]
+    public class AddAttributeAnnotationQuickFixTests : QuickFixTestBase
```
[the remark this answers is not in the document]
Comment (MDoerner) [reply RubberduckTests/QuickFixes/AddAttributeAnnotationQuickFixTests.cs]: I guess, you mean _annotation_, not _attribute_. The behaviour in presence of other annotations is basically already covered by the tests for the `AnnotationUpdater`, but I can add something here, too, if you want.

[the remark this answers is not in the document]
Comment (bclothier) [reply RubberduckTests/QuickFixes/AddAttributeAnnotationQuickFixTests.cs]: Let me rephrase that. The case would be adding an annotation for a 2nd attribute. 

Example:
```
Attribute VB_PredeclaredID = True
'@PredeclareId
Attribute VB_Exposed = True
```

=> 

```
'@Exposed
Attribute VB_PredeclaredID = True
'@PredeclareId
Attribute VB_Exposed = True
```

We want to ensure that even when the existing annotations and attributes are in place, they are not affected by addition of a new annotation for one attribute lacking the corresponding annotation.

[the remark this answers is not in the document]
Comment (MDoerner) [reply RubberduckTests/QuickFixes/AddAttributeAnnotationQuickFixTests.cs]: done
